File: 0456 - 132 Pattern/solution.cpp
class Solution {
public:
    bool find132pattern(vector<int>& A) {
        int n = A.size(), s3 = INT_MIN, stk_idx = n;
        for(int i = n-1; i >= 0; i--){
            if(A[i] < s3) return true;
            while(stk_idx<n && A[stk_idx]<A[i]) s3 = A[stk_idx++];
            A[--stk_idx] = A[i];
        }
        return false;
    }
};

File: 0456 - 132 Pattern/solution.py
class Solution:
    def find132pattern(self, A: List[int]) -> bool:
        s3, stk = -inf, []
        for i in A[-1::-1]:
            if i < s3:
                return True
            while stk and stk[-1] < i:
                s3 = stk.pop()
            stk.append(i)
        return False